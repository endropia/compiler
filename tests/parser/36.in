var a: Integer = 1;

begin
if a > 1 then a := 2;

if (a = 1) and (2 > 1) then a := 3 + 7 else a := 8;

if (a = 1) and (2 > 1) then a := 3 + 7 else
 begin
 a := 8;
 writeln(a);
 end;
end.