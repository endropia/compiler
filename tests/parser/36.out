program : Unnamed program
   var: 
      a
      type: Integer
      1
   stmts:
      if
         >
            a
            1
         :=
            a
            2
      if
         and
            =
               a
               1
            >
               2
               1
         :=
            a
            +
               3
               7
         else
            :=
               a
               +
                  3
                  7
      if
         and
            =
               a
               1
            >
               2
               1
         :=
            a
            +
               3
               7
         else
            :=
               a
               +
                  3
                  7
